test/libdnf/system/test_state.cpp

```diff
@@ -63,6 +63,16 @@ enabled_stream = "stream-1"
 
 )"""};
 
+static inline void erase_newlines(std::string & str) {
+    str.erase(std::remove(str.begin(), str.end(), '\n'), str.cend());
+}
+
+static void assert_equal_without_newlines(std::string expected, std::string actual) {
+    erase_newlines(expected);
+    erase_newlines(actual);
+
+    CPPUNIT_ASSERT_EQUAL(expected, actual);
```

Comment:
The issue with doing this is that whenever you get a failure, you'll get the line number of this `CPPUNIT_ASSERT_EQUAL()` call, and you'll be quite clueless which one is actually failing.

I'd return a trimmed copy from a `trimmed(...)` function and call that function on the file contents inside all the ASSERTs.